testing/generate.js

```diff
@@ -23,18 +26,35 @@ async function globalSetup() {
   await page.waitForSelector('.app-browser', {timeout: 180000}) // Wait three minutes for apps to sync
   await page.context().storageState({ path: 'state.json' })
   await browser.close()
+
+  fs.mkdirSync(resultsDir)
 }
 
 export default globalSetup
 `
 }
 
+function globalTeardown() {
```

Comment:
My first inclination was to use an `afterAll` hook to each test but that Playwright does not execute the `afterAll` hook if any of the tests in the group fails. AFAIK, there's no way to avoid that.